packages/core-plugin-api/src/apis/definitions/LogApi.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApiRef, createApiRef } from '../system';
+import { useApi } from '../hooks';
+import { LogLevel } from './LogStoreApi';
+
+/**
+ * A wrapper for the fetch API, that has additional behaviors such as the
+ * ability to automatically inject auth information where necessary.
+ *
+ * @public
+ */
+export type LogApi = {
+  [Level in LogLevel]: (...args: any) => void;
+};
+
+/**
+ * The {@link ApiRef} of {@link LogApi}.
+ *
+ * @public
+ */
+export const logApiRef: ApiRef<LogApi> = createApiRef({
+  id: 'core.log',
+});
+
+/**
+ * Utility wrapper around `useApi(logApiRef)` which decorates logging with the
```

Comment:
Just to highlight this, I think this is perhaps the best way to go about the decorating the log API with a plugin, even though that's not what the implementation is atm xP

Reply:
Although then again that makes it tricky for API created through factories. Tbh a log API was the one really strong use-case for plugin specific APIs, but really the only one in the end. It may be worth making some exception and building per-plugin log API implementations into the core instead, rather than making them a platform feature

Reply:
You're probably right, making it a core first class citizen might make sense, but, consider the usefulness of being able to swap it for something else (maybe something that wraps the core impl) by overriding the api.
And, there are some other places where similar functionalities should be very core, but overridable and maybe with the plugin/api context injected, namely Fetch. Or any other comms interface. And maybe analytics, metrics, ....

As discussed separately, the tricky thing is exposing internal "details"-ish while not needing to change it later (because head ache) and not encouraging over-using it...

The thinking with the logging is that you could swap them out, but perhaps that's not necessary. If the LogStoreApi could take a set of "transports" though (like winston) to fan out to, that'd fill most use-cases.